tiatoolbox/tools/registration/wsi_registration.py

```diff
@@ -539,3 +542,314 @@ def estimate_affine_transform(
+
+    def register(
+        self,
+        fixed_img: np.ndarray,
+        moving_img: np.ndarray,
+        fixed_mask: np.ndarray,
+        moving_mask: np.ndarray,
+    ) -> np.ndarray:
+        """Image Registration.
+
+        This function aligns a pair of images using Deep
+        Feature based Registration (DFBR) method.
+
+        Args:
+            fixed_img (:class:`numpy.ndarray`):
+                A fixed image.
+            moving_img (:class:`numpy.ndarray`):
+                A moving image.
+            fixed_mask (:class:`numpy.ndarray`):
+                A binary tissue mask for the fixed image.
+            moving_mask (:class:`numpy.ndarray`):
+                A binary tissue mask for the moving image.
+
+        Returns:
+            :class:`numpy.ndarray`:
+                An affine transformation matrix.
+
+        """
+        (
+            fixed_tissue,
+            fixed_mask,
+            moving_tissue,
+            moving_mask,
+            tissue_top_left_coord,
+        ) = self.get_tissue_regions(fixed_img, fixed_mask, moving_img, moving_mask)
+        features = self.extract_features(fixed_tissue, moving_tissue)
+        fixed_matched_points, moving_matched_points, quality = self.feature_mapping(
+            features
+        )
+        (
+            fixed_matched_points,
+            moving_matched_points,
+            quality,
+        ) = self.filtering_matching_points(
+            fixed_mask,
+            moving_mask,
+            fixed_matched_points,
+            moving_matched_points,
+            quality,
+        )
+
+        tissue_transform = DFBRegister.estimate_affine_transform(
+            fixed_matched_points, moving_matched_points
+        )
+        moving_tissue = cv2.warpAffine(
+            moving_tissue, tissue_transform[0:-1][:], fixed_tissue.shape[:2][::-1]
+        )
+        moving_mask = cv2.warpAffine(
+            moving_mask, tissue_transform[0:-1][:], fixed_tissue.shape[:2][::-1]
+        )
+
+        blocks_bbox = [
+            [
+                0,
+                np.floor(fixed_tissue.shape[0] / 2),
+                0,
+                np.floor(fixed_tissue.shape[1] / 2),
+            ],
+            [
+                0,
+                np.floor(fixed_tissue.shape[0] / 2),
+                np.ceil(fixed_tissue.shape[1] / 2),
+                fixed_tissue.shape[1],
+            ],
+            [
+                np.ceil(fixed_tissue.shape[0] / 2),
+                fixed_tissue.shape[0],
+                0,
+                np.floor(fixed_tissue.shape[1] / 2),
+            ],
+            [
+                np.ceil(fixed_tissue.shape[0] / 2),
+                fixed_tissue.shape[0],
+                np.ceil(fixed_tissue.shape[1] / 2),
+                fixed_tissue.shape[1],
+            ],
+        ]
+        fixed_matched_points, moving_matched_points, quality = [], [], []
+        for _index, block in enumerate(blocks_bbox):
+            bbox = [int(x) for x in block]
```

Comment:
```suggestion
            bounding_box = [int(x) for x in block]
```

Reply:
also shouldn't the block should already be defined as int to avoid an errors?

Reply:
https://github.com/TissueImageAnalytics/tiatoolbox/blob/7be8b4c7387ead7ee0cf082f14ded5ab87bc771a/tiatoolbox/tools/registration/wsi_registration.py#L776-L804